### PR TITLE
Send index immediately on local change

### PR DESCRIPTION
**Pull request by calmh**
This removes the delay between local index update and index send, by listening for the relevant events. It at the same time reduces the amount of polling we do.

### CODE REVIEW

**Comment by Zillode** [comment]
Cool

**Comment by AudriusButkevicius** [comment]
Have you verified this works? It's a bit hard to follow due to how the subscription is consumed and thrown away. I guess there is no point to check the event arguments instead of using m.LocalVersion?

**Comment by calmh** [comment]
It passes integration at least. :) We want to check the local version anyway, because things may have changed between us getting the event and processing it and so on. My first idea was to not do the whole database scan thing that sendIndexTo does and just send the update over a channel directly from updateLocals, but there are some details around that that need ironing out.,,

**Comment by calmh** [comment]
@st-jenkins run integration tests

**Comment by AudriusButkevicius** [comment]
So I am happy to merge this

**Comment by Zillode** [comment]
@st-jenkins run on windows please

**Comment by Zillode** [comment]
Hmm, I hope https://github.com/syncthing/syncthing/commit/404c08bfb10d4e650b0826b7460807e11ee9a327 fixes the integration tests for windows, but I'm off for now. If that patch works, maybe cherry-pick that one in here?

**Comment by AudriusButkevicius** [comment]
No merge?

**Comment by calmh** [comment]
Yes, is merged. I just did it sneakily. :)

**Comment by AudriusButkevicius** [comment]
No it's not.

**Comment by calmh** [comment]
You're right, I just pushed it to my own repo, not the real one. Fixed :dancers:
